extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -113,4 +110,32 @@ public void close()
     unions.clear();
   }
 
+  @Override
+  public void relocate(int oldPosition, int newPosition, ByteBuffer newBuffer)
+  {
+    NativeMemory nm = getNativeMemory(newBuffer);
+
+    Memory mem = new MemoryRegion(nm, newPosition, maxIntermediateSize);
+    Union newUnion = (Union) SetOperation.wrap(mem);
+
+    Union union = unions.get(oldPosition);
+    if (union != null) {
+      unions.remove(oldPosition);
+      unions.put(newPosition, newUnion);
+    } else {
+      unions.put(newPosition, newUnion);
+    }
+
+  }
+
+  private NativeMemory getNativeMemory(ByteBuffer buffer)
+  {
+    NativeMemory nm = nmCache.get(buffer);
```

Comment:
- More readable, intent is clearer: computeIfAbsent speaks for itself, harder to make a mistake.
 - One operation with Map instead of two separate operations.